test/tests/ide/ssh/ssh_gateway_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package ide
+
+import (
+	"context"
+	"io"
+	"log"
+	"net/url"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/helloyi/go-sshclient"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	gitpod "github.com/gitpod-io/gitpod/gitpod-protocol"
+	"github.com/gitpod-io/gitpod/test/pkg/integration"
+)
+
+func TestSSHGatewayConnection(t *testing.T) {
+	userToken, _ := os.LookupEnv("USER_TOKEN")
+	integration.SkipWithoutUsername(t, username)
+	integration.SkipWithoutUserToken(t, userToken)
+
+	f := features.New("TestSSHGatewayConnection").
+		WithLabel("component", "server").
+		Assess("it can connect to a workspace via SSH gateway", func(_ context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+			defer cancel()
+
+			api := integration.NewComponentAPI(ctx, cfg.Namespace(), kubeconfig, cfg.Client())
+			t.Cleanup(func() {
+				api.Done(t)
+			})
+
+			_, err := api.CreateUser(username, userToken)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			serverOpts := []integration.GitpodServerOpt{integration.WithGitpodUser(username)}
+			server, err := api.GitpodServer(serverOpts...)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			_ = server.SetEnvVar(ctx, &gitpod.UserEnvVarValue{
```

Comment:
maybe add a comment with a link to a regression PR or an incident for the future